include/nbl/builtin/hlsl/bxdf/fresnel.hlsl

```diff
@@ -115,33 +172,201 @@ struct refract
         return backside ? absNdotT : -(absNdotT);
     }
 
-    vector_type doRefract()
+    vector_type operator()()
     {
         return N * (NdotI * rcpOrientedEta + computeNdotT(backside, NdotI2, rcpOrientedEta2)) - rcpOrientedEta * I;
     }
 
-    static vector_type doReflectRefract(bool _refract, NBL_CONST_REF_ARG(vector_type) _I, NBL_CONST_REF_ARG(vector_type) _N, scalar_type _NdotI, scalar_type _NdotTorR, scalar_type _rcpOrientedEta)
+    vector_type I;
+    vector_type N;
+    bool backside;
```

Comment:
you want an `NdotT` member instead of `backside` and a `computeNdotT(const bool backside)` **non-static** method which can recompute it for you

Reply:
you also need a `computeNdotI` method